Automattic/newspack-ios
# Issues/117 enable video part 3

Closes #117 

This is the third of a series of three PRs introducing video support to the app. 

In this PR we introduce a video details screen, and wire up a thumbnail for the video cell.  This is done by replacing the PhotoDetailsViewController with a MediaDetailsViewController that serves both photos and video asset types.  Image creation is also improved by introducing a helper class in the Newspack Framework that is now responsible for image creation for photo and video types in both the share extension and the main app.

Note: The "play" icon in the video cells is currently off-center.  We'll address this in future UI polish. 

To test: 
- Confirm video cells show a thumbnail of the video.
- Smoke test the photos feature and the functionality in the photo details screen.  Ensure everything still works as expected.
- Perform the same tests with a video. Ensure you can modify video properties, see a preview image of the video, and play the video by tapping on the image. 
- Check the share extension and make sure thumbnails are still showing up for images and video. 

@jleandroperez Hi hi 👋.  Game for another one?  No hurry at all sir.
